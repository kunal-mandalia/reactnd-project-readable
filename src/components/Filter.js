import React, {Component} from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { filterByCategory } from '../actions/index'

export class Filter extends Component {
  render () {
    const { filter, categories, filterByCategory } = this.props
    return (
      <div className='filter'>
        {categories.map(c => (
          <div key={c.path}>
            <div
              className={`category ${filter === c.path ? 'active' : 'inactive'}`}
              onClick={filterByCategory.bind(null, c.path)}>{c.name}</div>
          </div>
        ))}
      </div>
    )
  }
}

Filter.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.object).isRequired,
  filterByCategory: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired
}

const mapStateToProps = state => ({
  filter: state.filter
})

const mapDispatchToProps = dispatch => ({
  filterByCategory: category => dispatch(filterByCategory(category))
})

export default Filter
