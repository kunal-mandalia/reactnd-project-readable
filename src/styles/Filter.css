.filter {
  display: inline;
}

.filter-category {
  cursor: pointer;
}

.filter-category.active {
  padding-bottom: 2px;
  border-bottom: solid 4px white;
}

.filter-category.inactive:hover {
  padding-bottom: 2px;
  border-bottom: solid 4px paleturquoise;
}

.filter-separator {
  color: rgba(255, 255, 255, 0.3);
}
