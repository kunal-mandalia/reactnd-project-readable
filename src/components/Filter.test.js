import React from 'react'
import { Filter } from './Filter'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

const mockFnOnFilter = jest.fn()
const props = {
  categories: [
    {
      name: 'react',
      path: 'react'
    },
    {
      name: 'redux',
      path: 'redux'
    },
    {
      name: 'udacity',
      path: 'udacity'
    }
  ],
  filterByCategory: mockFnOnFilter,
  filter: ``
}

describe(`<Filter .../>`, () => {
  it(`should render ${props.categories.length} categories`, () => {
    const wrapper = shallow(<Filter {...props} />)
    expect(wrapper.find('.category')).toHaveLength(props.categories.length)
  })

  it(`should call filterByCategory when clicking on a category`, () => {
    const wrapper = shallow(<Filter {...props} />)
    wrapper.find('.category').first().simulate('click')
    expect(mockFnOnFilter.mock.calls).toHaveLength(1)    
  })

  it(`should match snapshot`, () => {
    const tree = renderer.create(<Filter {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
