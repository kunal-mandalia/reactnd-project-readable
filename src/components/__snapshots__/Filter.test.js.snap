// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter .../> should match snapshot 1`] = `
<div
  className="filter"
>
  <div>
    <div
      className="category inactive"
      onClick={[Function]}
    >
      react
    </div>
  </div>
  <div>
    <div
      className="category inactive"
      onClick={[Function]}
    >
      redux
    </div>
  </div>
  <div>
    <div
      className="category inactive"
      onClick={[Function]}
    >
      udacity
    </div>
  </div>
</div>
`;
